### PR TITLE
[REF] models: improve code of create/write

### PR DESCRIPTION
**Pull request by rco-odoo**
This is essentially some code cleanup and a few optimizations.

### CODE REVIEW

File: odoo/models.py [hunk 1]
```diff
@@ -3254,200 +3161,290 @@ def create(self, vals):
...
-        old_vals, new_vals, unknown = {}, {}, []
+        # distribute fields into sets for various purposes
+        store_vals = {}
+        inverse_vals = {}
+        inherited_vals = defaultdict(dict)      # {modelname: {fieldname: value}}
+        unknown_names = []
+        protected_fields = []
         for key, val in vals.items():
+            if key in bad_names:
+                continue
```
Comment (kujiu) [inline odoo/models.py]: Perhaps log a warning, the dev may think he can override values...

File: odoo/models.py [hunk 2]
```diff
@@ -41,6 +41,7 @@
 import psycopg2
 from lxml import etree
 from lxml.builder import E
+from psycopg2.extensions import AsIs
```
Comment (hmo-odoo) [inline odoo/models.py]: can not use sql.Identifier method of psycopg2.sql module   ?

Comment (rco-odoo) [reply odoo/models.py]: You are right, we should eventually use `psycopg2.sql.Identifier` for formatting table and column names. But that is not the point of this pull request. I would do a separate cleanup with such improvements in all places where we compose SQL queries programmatically.

**Comment by odony** [comment]
- test warning for commit 262519c1ad42ac0af2d3611b75c28fe08e27a2e9 (deps on inherited fields), to see if we have many cases in practice, or if the declaration should be fixed (+ if we keep this, add a regression test for this case)
- maybe rewrite the `for` loops on _inherits in 6f0598a779e4dce14a385797f1cabc1cbf6caf07 to avoid the `continue` lines
- test import of e.g. 40k stock locations, before taking any decision wrt to parent store compute deferral

**Comment by rco-odoo** [comment]
Parent store compute deferral seems necessary.  Importing 22k stock locations gives
 - with deferral: 9 minutes,
 - without deferral: more than 30 minutes.

**Pull request merged**